src/magic_folder/config.py

```diff
@@ -1269,13 +1259,14 @@ def rotate_api_token(self):
         return self._token_provider.rotate()
 
     @property
-    @with_cursor
```

Comment:
It seems a little bit unfortunate that `with_cursor` is now incompatible with `GlobalConfigDatabase`?  Maybe mainly because this is a surprising outcome - only *some* of our database code can use the helper.  But maybe also eventually practically, as we have to duplicate more transaction management code?  Or forget to duplicate it and just get it wrong.

Is it already wrong?  `remove_magic_folder` looks up the location of folders separately from deleting them.  And little of the code in `GlobalConfigDatabase` explicitly starts a transaction at all.  There's a lot of `with self.database` but I don't remember what that means for a pysqlite Connection object.

Reply:
I agree. The reason I made it incompatible with the current `GlobalConfigDatabase` code is because I wanted to ensure that any code that touches the same database as `with_cursor` goes through the same code, to ensure that we have consistent transaction handling.

There was enough code in `GlobalConfigDatabase` that didn't use `with_cursor` that I didn't want to rewrite entirely in this PR, that I thought it made more sense to remove the two uses of it here, for the moment.

>  There's a lot of with self.database but I don't remember what that means for a pysqlite Connection object.

All it does is call either `.commit` or `.rollback` when the block exits, depending on whether the block exits with an exception or not. However, [by default](https://docs.python.org/2.7/library/sqlite3.html#controlling-transactions), pysqlite automatically starts a transaction for any statement that it thinks modifies data.

> Is it already wrong? [...]

Skimming the code, I don't *think* so, assuming that we are the only process writing to the database (which we should be, assuming there isn't multiple copies of the service running, which is an assumption we make elsewhere too).

Reply:
In fact, it looks like I have a local commit switching it over, but it was causing test failures, due to `list_magic_folders` being a generator, which is used in a context where `get_magic_folder` is called while iterating the generator. That is certainly fixable, but I didn't want spend time on it, particularly given that both of those functions are read-only.